### PR TITLE
fix(datepicker): reduced opacity to set disabled state for calendar icon

### LINKED ISSUE #15983: [Bug]: Datepicker & Datepicker range calendar icon in disabled state
### Package

@carbon/react

### Browser

Chrome

### Package version

1.53.0

### React version

18.2.0

### Description

Calendar icon doesn't look disabled in Datepicker & Datepicker range components -

Datepicker -
Screenshot 2024-03-15 at 3 55 31 PM

Datepicker range - 
Screenshot 2024-03-15 at 3 54 39 PM

### Reproduction/example

https://stackblitz.com/edit/github-vvajoh?file=src%2FApp.jsx

### Steps to reproduce

Please see reproduction example above.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by anjaly0606**
Closes   
#15983 

Added disabled state for datepicker and datepicker range calendar icon by icon disabled fill 

#### Changelog

**Changed**

- _date-picker.scss 
fill icon disabled

#### Testing / Reviewing

{{ Datepicker component }}

### CODE REVIEW

**Comment by anjaly0606** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/styles/scss/components/date-picker/_date-picker.scss [hunk 1]
```diff
@@ -177,6 +177,7 @@
 
   .#{$prefix}--date-picker__input:disabled ~ .#{$prefix}--date-picker__icon {
     cursor: not-allowed;
+    opacity: 0.3;
```
Comment (tw15egan) [inline packages/styles/scss/components/date-picker/_date-picker.scss]:
```suggestion
    fill: $icon-disabled;
```

Can we use the disabled icon token instead?

**Review by tw15egan — CHANGES_REQUESTED** [review]

File: packages/styles/scss/components/date-picker/_date-picker.scss (hunk 1 above)
Comment (anjaly0606) [reply packages/styles/scss/components/date-picker/_date-picker.scss]: Sure

**Comment by anjaly0606** [comment]
recheck

**Review by tw15egan — APPROVED** [review]

**Comment by tw15egan** [comment]
@anjaly0606, it seems your local git config username does not match your public github username, hence the failed DCO checks 

Screenshot 2024-03-21 at 2 02 24 PM

**Comment by anjaly0606** [comment]
Closing this PR as there is a conflict with the local git username, Will raise a new one

**Pull request closed without merging**